### PR TITLE
99suse-initrd: Exclude modules that are built-in

### PR DESCRIPTION
**Pull request by leberus**
When checking the required modules a certain module needs, we need
to check whether those are already built-in or not.
If they are, we exclude them from the list of modules to install.

This is necessary as otherwise dracut-install fails when stumbling
over modules thar are already built-in.

Signed-off-by: Oscar Salvador <osalvador@suse.de>

This pull request changes...

## Changes

We check whether a certain module is built-in before adding it to the list
of modules to install.

## Checklist
- [X] I have tested it locally
- [X] I have reviewed and updated any documentation if relevant
- [X] I am providing new code and test(s) for it

Fixes #

When using "--no-hostonly", dracut will fail when executing "dracut: *** Including module: suse-initrd ***" step
because some modules it tries to find, are actually built-in.
E.g:

```
dracut: *** Including module: suse-initrd ***
dracut-install: ERROR: installing 'ecb'
dracut: FAILED: /usr/lib/dracut/dracut-install -D /var/tmp/dracut.CQM5Ve/initramfs -N i2o_scsi --kerneldir /lib/modules/5.3.18-24.61-default/ -m ehci-hcd ehci-hcd crc32c des ecb md4 md5 hmac arc4 nls_utf8 cxgb4
```

### CODE REVIEW

File: modules.d/99suse-initrd/module-setup.sh [hunk 1]
```diff
@@ -22,14 +22,23 @@ get_modprobe_conf_files() {
...
         mod="${line##*SUSE INITRD: }"
         mod="${mod%% REQUIRES*}"
         reqs="${line##*REQUIRES }"
         if [[ ! $hostonly ]] || grep -q "^$mod\$" "$DRACUT_KERNEL_MODALIASES"
         then
-            all_mods="$all_mods $reqs"
+            for module in $reqs;
+            do
+                is_built_in=$(awk 'BEGIN { FS = "/" } ; { print $NF }' $BUILT_IN_PATH | grep -wc $module.ko)
+                if [ $is_built_in -eq 0 ]
```
Comment (danimo) [inline modules.d/99suse-initrd/module-setup.sh]: How about:

`if grep -q "/$module.ko" "$BUILT_IN_PATH"; then ...`

which does not require awk (needs to be checked for and installed) nor use an (expensive) subshell? If you really want this to be thorough, you can use `grep -qe "/$module.ko$^"` to be extra-safe.

**Review by danimo — CHANGES_REQUESTED** [review]
One minor thing, LGTM otherwise.

File: modules.d/99suse-initrd/module-setup.sh (hunk 1 above)
Comment (leberus) [reply modules.d/99suse-initrd/module-setup.sh]: Looks good to me, should be the following though:

`if ! grep -q "/$module.ko" $BUILT_IN_PATH`

as we are only interested in those that are not built-in.

Thanks for the review

**Comment by leberus** [comment]
Changes applied.

**Review by danimo — APPROVED** [review]

**Comment by danimo** [comment]
Please change the commit subject to "fix(suse-initrd): exclude modules that are built-in " to make commisery happy.

**Pull request merged**